torchao/quantization/quantize_/workflows/float8/float8_tensor.py

```diff
@@ -248,14 +248,151 @@ def from_hp(
 implements_torch_function = Float8Tensor.implements_torch_function
 
 
-@implements([aten.linear.default])
-@implements_torch_function([torch.nn.functional.linear])
+@implements(aten.linear.default)
+@implements_torch_function(torch.nn.functional.linear)
 def _(func, types, args, kwargs):
     input_tensor, weight_tensor, bias = (
         args[0],
         args[1],
         args[2] if len(args) > 2 else None,
     )
+    return _float8_linear_impl(input_tensor, weight_tensor, bias)
+
+
+@implements(aten.matmul.default)
+@implements_torch_function(torch.matmul)
+def _(func, types, args, kwargs):
+    input_tensor, weight_tensor = args[0], args[1]
+    return _float8_matmul_impl(input_tensor, weight_tensor)
+
+
+@implements(aten.mm.default)
+@implements_torch_function(torch.mm)
+def _(func, types, args, kwargs):
+    input_tensor, weight_tensor = args[0], args[1]
+    return _float8_matmul_impl(input_tensor, weight_tensor)
+
+
+@implements(aten.addmm_.default)
+def _(func, types, args, kwargs):
+    bias_tensor, input_tensor, weight_tensor = (
+        args[0],
+        args[1],
+        args[2],
+    )
+    assert kwargs.get("alpha", 1) == 1, "only alpha=1 is supported"
+    assert kwargs.get("beta", 1) == 1, "only beta=1 is supported"
+    out = _float8_matmul_impl(input_tensor, weight_tensor)
+    return bias_tensor.add_(out)
+
+
+def _get_matmul_kernel_choice(weight_tensor: Float8Tensor) -> str:
+    """
+    Return the kernel choice for matmuls, either "fbgemm" or "torch".
+    """
+    if weight_tensor.kernel_preference == KernelPreference.AUTO:
+        kernel_choice = "torch"
+        if _is_fbgemm_gpu_genai_available() and is_sm_at_least_90():
+            kernel_choice = "fbgemm"
+    elif weight_tensor.kernel_preference == KernelPreference.FBGEMM:
+        kernel_choice = "fbgemm"
+    else:
+        assert weight_tensor.kernel_preference == KernelPreference.TORCH, (
+            f"{weight_tensor.kernel_preference=} not handled"
+        )
+        kernel_choice = "torch"
+    return kernel_choice
+
+
+def _call_fbgemm_f8f8bf16_matmul(
+    input_tensor: Float8Tensor,
+    weight_tensor: Float8Tensor,
+    bias: Optional[torch.Tensor] = None,
+    weight_is_already_transposed: bool = False,
```

Comment:
instead of this flag, just transpose at the callsite to match the meaning of `matmul`